Applications/spire/source/charting/chart_view.cpp

```diff
@@ -0,0 +1,158 @@
+#include "spire/charting/chart_view.hpp"
+#include <locale>
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <QFontMetrics>
+#include <QPainter>
+#include <QPaintEvent>
+#include "Nexus/Definitions/Money.hpp"
+#include "spire/spire/dimensions.hpp"
+
+using namespace boost::posix_time;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  QVariant to_variant(ChartValue::Type type, ChartValue value) {
+    if(type == ChartValue::Type::DURATION) {
+      return QVariant::fromValue(static_cast<time_duration>(value));
+    } else if(type == ChartValue::Type::MONEY) {
+      return QVariant::fromValue(static_cast<Money>(value));
+    } else if(type == ChartValue::Type::QUANTITY) {
+      return QVariant::fromValue(static_cast<Quantity>(value));
+    } else if(type == ChartValue::Type::TIMESTAMP) {
+      return QVariant::fromValue(static_cast<ptime>(value));
+    }
+    return QVariant();
+  }
+}
+
+ChartView::ChartView(ChartValue::Type x_axis_type,
+    ChartValue::Type y_axis_type, QWidget* parent)
+    : QWidget(parent),
+      m_x_axis_type(x_axis_type),
+      m_y_axis_type(y_axis_type),
+      m_timestamp_format("%H:%M:%S") {
+  setFocusPolicy(Qt::NoFocus);
+  m_label_font = QFont("Roboto");
```

Comment:
Initialize values in the initializer list, ie. before line 34.

Reply:
Does this work for [QFont](http://doc.qt.io/qt-5/qfont.html)? The only other constructor I see takes point size, not pixel size.

Reply:
Yes exactly.